app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt

```diff
@@ -60,10 +74,62 @@ class RecipesViewModel(private val repository: RecipeRepository) : ViewModel() {
    */
   fun updateCurrentRecipe(recipe: Recipe) {
     _currentRecipe.value = recipe
+    updateNextRecipe()
   }
 
   /** Clears the current selected recipe. */
   fun clearCurrentRecipe() {
     _currentRecipe.value = null
   }
+
+  /** Gives the next recipe in the list of recipes. */
+  fun nextRecipe() {
```

Comment:
Thank you for your feedback. Changes has been provided for the 2 last points. However for the first one, it is way better to remove the first element at each swipe, so we dont have a big list if the user swipes a large amount of time. However it will be interesting to delete by groups of recipes and not each time that the user swipes. Thank you for your insight !!